### PR TITLE
Fix/auth merge

### PR DESCRIPTION
**Pull request by HenrikSZ**
- fixed remaining auth/login troubles
- fixed friend list types
- integrated jwt token refresh process
- reorganized navigators
- made text input text visible

### CODE REVIEW

File: frontend/app/authSlice.ts [hunk 1]
```diff
@@ -1,69 +1,65 @@
...
   let status: AuthStatus = 'loading';
-  let token: string = '';
+  let refreshToken: string = '';
   try {
     const credentials = await Keychain.getInternetCredentials('greengrub');
     if (credentials) {
-      token = credentials.password; 
-      status = 'authenticated';
+      refreshToken = credentials.password;
+      status = 'stale';
```
Comment (adelmuursepp) [inline frontend/app/authSlice.ts:40]: Don't know what stale is but will read on it, otherwise makes sense

File: frontend/app/MainTabs.tsx [hunk 2]
```diff
@@ -11,24 +11,24 @@ import {createBottomTabNavigator} from '@react-navigation/bottom-tabs';
...
-import ScanScreen from './scan/ScanScreen';
-import FriendsScreen from './friends/FriendsScreen';
+import ScanTab from './scan/ScanTab';
+import FriendsTab from './friends/FriendsTab';
 import ProfileScreen from './profile/ProfileScreen';
 
 export type MainTabsParamList = {
   Home: undefined;
-  Scan: undefined;
+  'Scan Tab': undefined;
```
Comment (adelmuursepp) [inline frontend/app/MainTabs.tsx:20]: Why just some of them are saves as strings? I get the error when trying to navigate to Saved Items that no such screen is available ("Saved Items" was not handled by any Navigator)

File: frontend/app/authSlice.ts (hunk 1 above)
Comment (HenrikSZ) [reply frontend/app/authSlice.ts:40]: Okay, will add docs to it. It is a state when the access token has expired. I didn't really use it, but it's there if we ever want to use it.

File: frontend/app/MainTabs.tsx (hunk 2 above)
Comment (HenrikSZ) [reply frontend/app/MainTabs.tsx:20]: They're basically all strings. You just need to add the quotes when there are spaces.

**Pull request merged**